app/src/main/java/com/example/androidchallenge/Data/ApiWeather.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.androidchallenge.Data
+import retrofit2.Call
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+
+object ApiWeather {
+    var BASE_URL:String="http://10.0.2.2:3000/"
+    val getWeather: Call<WeatherWeekJson>
+        get(){
+        val retrofit = Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        val service = retrofit.create(ApiInterface::class.java)
+        val call = service.getWeather()
+        return call
+    }
+    val getToday: Call<Weather>
+        get(){
+            val retrofit = Retrofit.Builder()
+                .baseUrl(BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build()
+            val service = retrofit.create(ApiInterface::class.java)
+            val call = service.getToday()
+            return call
+        }
+}
```

Comment:
Consider this as an extension of the comment given [here](https://github.com/niina1/android-challenge/pull/1/files#r633932563).

There are two things happening here that could be separated. I'll try to keep it simple...

1. For us to use Retrofit as a tool to fetch data from a HTTP API we first, net to create an instance of it (lines 11 and 21). With that instance, we can them create our APIs by providing the interfaces that describe their behaviour (lines 15 and 25). Retrofit will create the rest for us and we'd only need to worry about calling the methods. In short, one of the things this file is doing, is managing our Retrofit instances;

2. The other thing is simpler... you just calling our API's methods to get te information that you need! 

It's a good practice to keep this two things separated. Having said that I suggest:
1. Rename the object to `RetrofitHelper` and try to focus on only dealing with the retrofit creation and setup up. Note: You don't actually need two instances (lines 11 and 21) since you are using the same BASE_URL. 👍 
2. This would be a bit more complex: [here](https://github.com/niina1/android-challenge/pull/1/files#r633930695), I mentioned that you're try to handle a "data layer" operation on your presentation layer... there, you are actually accessing data from a remote service, we usually have "services classes" that does just that by handling the calls to the API's and any possible errors. Lines 15/16 and 25/26 could be moved to a, let's say, a `WeatherService` that joins everything all together! 👍